### PR TITLE
Add support for encoding 16 bit per pixel bitmaps

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

The PR add the ability to encode bitmaps with 16 bits per pixel.

### CODE REVIEW

File: src/ImageSharp/PixelFormats/PixelImplementations/Bgra5551.cs [hunk 1]
```diff
@@ -87,6 +87,10 @@ public Vector4 ToVector4()
                         (this.PackedValue >> 15) & 0x01);
         }
 
+        /// <inheritdoc />
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public void FromBgra5551(Bgra5551 source) => this.FromScaledVector4(source.ToScaledVector4());
```
Comment (antonfirsov) [inline src/ImageSharp/PixelFormats/PixelImplementations/Bgra5551.cs]: Could be `this = source` I guess.

**Comment by JimBobSquarePants** [comment]
The king of bmp strikes again! 

Looks like our pixel format coverage has dropped significantly. We'll have to update the individual tests. It's a chore of a job sorry!

File: src/ImageSharp/PixelFormats/PixelImplementations/Bgra5551.cs (hunk 1 above)
Comment (brianpopow) [reply src/ImageSharp/PixelFormats/PixelImplementations/Bgra5551.cs]: i have changed that

File: src/ImageSharp/PixelFormats/PixelImplementations/Rgb24.cs [hunk 2]
```diff
@@ -156,6 +156,10 @@ public void FromGray16(Gray16 source)
             this.B = rgb;
         }
 
+        /// <inheritdoc />
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public void FromBgra5551(Bgra5551 source) => this.FromVector4(source.ToVector4());
```
Comment (JimBobSquarePants) [inline src/ImageSharp/PixelFormats/PixelImplementations/Rgb24.cs]: `FromScaledVector4`, `FromScaledVector4`.

**Pull request merged**